x-pack/packages/kbn-cloud-security-posture/graph/src/components/graph/graph.tsx

```diff
@@ -66,28 +81,45 @@ const edgeTypes = {
  *
  * @returns {JSX.Element} The rendered Graph component.
  */
-export const Graph: React.FC<GraphProps> = ({ nodes, edges, interactive, ...rest }) => {
-  const layoutCalled = useRef(false);
-  const [isGraphLocked, setIsGraphLocked] = useState(interactive);
-  const { initialNodes, initialEdges } = useMemo(
-    () => processGraph(nodes, edges, isGraphLocked),
-    [nodes, edges, isGraphLocked]
-  );
-
-  const [nodesState, setNodes, onNodesChange] = useNodesState(initialNodes);
-  const [edgesState, _setEdges, onEdgesChange] = useEdgesState(initialEdges);
-
-  if (!layoutCalled.current) {
-    const { nodes: layoutedNodes } = layoutGraph(nodesState, edgesState);
-    setNodes(layoutedNodes);
-    layoutCalled.current = true;
-  }
+export const Graph: React.FC<GraphProps> = ({
+  nodes,
+  edges,
+  interactive,
+  isLocked: isLockedProp = false,
+  ...rest
+}) => {
+  const backgroundId = Math.random().toFixed(4); // TODO: use useId(); when available (react >=18)
+  const fitViewRef = useRef<
+    ((fitViewOptions?: FitViewOptions<Node> | undefined) => Promise<boolean>) | null
+  >(null);
+  const [prevNodes, setPrevNodes] = useState<NodeViewModel[]>([]);
+  const [prevEdges, setPrevEdges] = useState<EdgeViewModel[]>([]);
+  const [isGraphLocked, setIsGraphLocked] = useState(!interactive);
```

Comment:
It seems confusing that it uses `interactive` instead of `isLocked` in this method, what about renaming the state to `isGraphInteractive` in that case?